NEWS.md

```diff
@@ -97,7 +97,9 @@ Command-line option changes
 
 Multi-threading changes
 -----------------------
-
+* Parts of the multi-threading API are now considered stable, with caveats.
+  This includes all identifiers exported from `Base.Threads` except the
```

Comment:
```suggestion
  This includes all documented identifiers from `Base.Threads` except the
```

I think it's better to not restrict to exported identifier because we may have `Threads.foreach` #34543 (or even `Threads.map`?).  As we are not going to export `Iterators.map` https://github.com/JuliaLang/julia/pull/34352#discussion_r365980331, it makes sense to not do this in `Base.Threads` as well.